docs/hello.rst

```diff
@@ -109,32 +108,35 @@ Initialize Taichi according to your hardware platform as follows:
 Tensors
 -------
 
-Taichi is a data-oriented programming language where **tensors** are the first-class citizens.
+Taichi is a data-oriented programming language where dense or spatially-sparse tensors are the first-class citizens.
+
+See :ref:`sparse` for more details on sparse tensors.
 
 In the code above, ``pixels = ti.var(dt=ti.f32, shape=(n * 2, n))`` allocates a 2D dense tensor named ``pixels`` of
-size ``(640, 320)`` with data type ``ti.f32`` (32-bit floating-point number).
+size ``(640, 320)`` and element data type ``ti.f32`` (i.e. ``float`` in C).
 
 
 Functions and kernels
 ---------------------
 
-Computation resides in Taichi **kernels**. Kernel arguments must be type-hinted.
-The language used in Taichi kernels and functions looks much like Python, yet the Taichi frontend compiler converts it
+Computation resides in Taichi **kernels**, which is defined with the decorator ``@ti.kernel``.
+Kernel arguments must be type-hinted (if any).
+The language used in Taichi kernels and functions looks exactly like Python, yet the Taichi frontend compiler converts it
 into a language that is **compiled, statically-typed, lexically-scoped, parallel and differentiable**.
 
-Taichi **functions**, which can be called by Taichi kernels and other Taichi functions, should be defined with the keyword ``ti.func``.
+Taichi **functions** are defined with the decorator ``@ti.func``.
+They can be called by Taichi kernels and other Taichi functions.
```

Comment:
Nit: you could join the above 2 sentences with 'and' and replace the 'and' there with 'or'.